[Withings] 'consumerKey' configuration key is unknown

Binding throws the following exeption:
```
2017-08-21 00:08:45.350 [ERROR] [org.apache.felix.configadmin        ] - [org.osgi.service.cm.ManagedService, id=1171, bundle=217/file:/usr/share/openhab2/addons/org.openhab.binding.withings-1.10.0.jar]: Updating property consumerKey of configuration org.openhab.withings-oauth caused a problem: The given configuration key is unknown!
```
## Expected Behavior
After trowing "org.openhab.binding.withings-1.10.0.jar" to the addons-folder this binding should setup itself.

## Possible Solution / Context
* I am not sure about the reasons. I tried to restart the binding.
Found that withings is now NOKIA. But URLs used in binding (WithingsAuthenticator.java) are still alive.
Minimum there should be an Update of the URLs
* I also tried to use the steps here [NOKIA-Health-API](https://developer.health.nokia.com/api) with DEFAULT_CONSUMER_KEY and DEFAULT_CONSUMER_SECRET given in WithingsAuthenticator.java.
Was able to find my withings-data when followed every step.
* Tried also to build my own **withings.cfg** like that (used ***** for privacy reasons):
```
token=04d08e***********************39fdcd06b4fe054c04
tokensecret=02c3fb2*********************de33c1cb1bc836bf774994c1
``` 
That gives the same error-messages
* Using the WIKI-Page was not very helpful because there are some contradictory informations. eg.:
`To change the OAuth parameters for the Withings binding the user can specify the following values in the services/withings.cfg:
The OAuth configuration must be stored in the file services/withings-oauth.cfg`

## Environment
* Version used: org.openhab.binding.withings-1.10.0.jar
* Operating System: openHABian (latest snapshot 2.2.0~20170820133825-1 (Build #1020)

## [DEBUG] log of start / stop
```
2017-08-21 00:08:43.581 [DEBUG] [org.openhab.binding.withings        ] - BundleEvent STARTING - org.openhab.binding.withings
2017-08-21 00:08:43.812 [DEBUG] [org.openhab.binding.withings        ] - BundleEvent STARTED - org.openhab.binding.withings
2017-08-21 00:08:44.376 [DEBUG] [org.openhab.binding.withings        ] - ServiceEvent REGISTERED - {org.osgi.service.cm.ManagedService}={service.pid=org.openhab.withings-oauth, osgi.command.function=[startAuthentication,finishAuthentication], component.name=org.openhab.binding.withings.authenticator, component.id=309, osgi.command.scope=withings, service.id=1171, service.bundleid=217, service.scope=bundle} - org.openhab.binding.withings
2017-08-21 00:08:45.350 [ERROR] [org.apache.felix.configadmin        ] - [org.osgi.service.cm.ManagedService, id=1171, bundle=217/file:/usr/share/openhab2/addons/org.openhab.binding.withings-1.10.0.jar]: Updating property consumerKey of configuration org.openhab.withings-oauth caused a problem: The given configuration key is unknown!
org.osgi.service.cm.ConfigurationException: consumerKey : The given configuration key is unknown!
	at org.openhab.binding.withings.internal.api.WithingsAuthenticator.updated(WithingsAuthenticator.java:251) [217:org.openhab.binding.withings:1.10.0]
	at org.apache.felix.cm.impl.helper.ManagedServiceTracker.updated(ManagedServiceTracker.java:189) [7:org.apache.felix.configadmin:1.8.14]
	at org.apache.felix.cm.impl.helper.ManagedServiceTracker.updateService(ManagedServiceTracker.java:152) [7:org.apache.felix.configadmin:1.8.14]
	at org.apache.felix.cm.impl.helper.ManagedServiceTracker.provideConfiguration(ManagedServiceTracker.java:85) [7:org.apache.felix.configadmin:1.8.14]
	at org.apache.felix.cm.impl.ConfigurationManager$ManagedServiceUpdate.provide(ConfigurationManager.java:1463) [7:org.apache.felix.configadmin:1.8.14]
	at org.apache.felix.cm.impl.ConfigurationManager$ManagedServiceUpdate.run(ConfigurationManager.java:1419) [7:org.apache.felix.configadmin:1.8.14]
	at org.apache.felix.cm.impl.UpdateThread.run0(UpdateThread.java:141) [7:org.apache.felix.configadmin:1.8.14]
	at org.apache.felix.cm.impl.UpdateThread.run(UpdateThread.java:109) [7:org.apache.felix.configadmin:1.8.14]
	at java.lang.Thread.run(Thread.java:745) [?:?]
2017-08-21 00:08:45.444 [DEBUG] [org.openhab.binding.withings        ] - ServiceEvent REGISTERED - {org.openhab.model.item.binding.BindingConfigReader, org.openhab.binding.withings.WithingsBindingProvider}={component.name=org.openhab.binding.withings.genericbindingprovider, component.id=311, service.id=1172, service.bundleid=217, service.scope=bundle} - org.openhab.binding.withings
2017-08-21 00:08:47.763 [DEBUG] [org.openhab.binding.withings        ] - ServiceEvent REGISTERED - {org.osgi.service.cm.ManagedService}={service.pid=org.openhab.withings, component.name=org.openhab.binding.withings.binding, component.id=310, service.id=1174, service.bundleid=217, service.scope=bundle} - org.openhab.binding.withings
```

--- a/bundles/binding/org.openhab.binding.withings/README.md
+++ b/bundles/binding/org.openhab.binding.withings/README.md
@@ -1,89 +1,123 @@
 # Withings Binding
 
-The Withings binding allows openHAB to synchronize data from the official Withings API to items. The following body measure types are supported: diastolic blood pressure, fat free mass, fat mass weight, fat ratio, heart pulse, height, systolic blood pressure, weight.
+The Withings binding allows openHAB to synchronize data from the official
+Withings API to items. The following body measure types are supported: diastolic
+blood pressure, fat free mass, fat mass weight, fat ratio, heart pulse, height,
+systolic blood pressure, and weight.
 
 ## Binding Configuration
 
 The binding can be configured in the file `services/withings.cfg`.
 
-| Property | Default | Required | Description |
-|----------|---------|:--------:|-------------|
-| refresh  | 360000  |    No    | The rate, in milliseconds, at which the binding requests Withings data. By default the Withings data is requested every 60 minutes (360000 milliseconds). |
-| consumerKey |      |    Yes   | your consumer key |
-| consumerSecret |   |    Yes   | your consumer secret |
-| redirectUrl |      |    Yes   | your redirect url |
+| Property       | Default | Required | Description |
+|----------------|---------|:--------:|-------------|
+| refresh        | 360000  |    No    | The periodicity, in milliseconds, at which the binding requests Withings data. By default the Withings data is requested every 60 minutes (360000 milliseconds). |
+| consumerkey    |         |    Yes   | your consumer key |
+| consumersecret |         |    Yes   | your consumer secret |
+| redirectUrl    |         |    Yes   | your redirect url |
 
-To access Withings data, the user needs to authenticate via an OAuth 1.0 flow. The binding implements the flow through the command line interface. The first time the binding is started, it prints the following messages to the console:
+### Configuration for multiple accounts
 
-```
+If using multiple withings accounts, the following properties are required
+for each account.
+
+| Property                   | Description |
+|----------------------------|-------------|
+| `<accountId>`.userid       | The userid for this account |
+| `<accountId>`.token        | The token for this account |
+| `<accountId>`.tokensecret  | The token secret for this account |
+
+
+To access Withings data, the user needs to authenticate via an OAuth 1.0 flow.
+The binding implements this flow through the command line interface. The first
+time the binding is started, it prints the following messages to the console:
+
+````
 #########################################################################################
 # Withings Binding needs authentication.
 # Execute 'withings:startAuthentication "<accountId>"' on OSGi console.
 #########################################################################################
-```
+````
 
-In order to start the authentication process, the user needs to execute `withings:startAuthentication <accountId>` on the OSGi console where `<accountId>` is an arbitrary key also used in the `services/withings-oauth.cfg` in order to differentiate between the different credentials per Withings account. 
+In order to start the authentication process, the user needs to execute
+`withings:startAuthentication <accountId>` on the OSGi console where
+`<accountId>` is an arbitrary key also used in the `services/withings.cfg`
+file in order to differentiate between the different credentials per Withings
+account.
 
 The binding will print the following lines to the console:
 
-```
+````
 #########################################################################################
 # Withings Binding Setup: 
 # 1. Open URL 'http://<auth-url>//' in your webbrowser
 # 2. Login, choose your user and allow openHAB to access your Withings data
 # 3. Execute 'withings:finishAuthentication "<accountId>" "<verifier>" "<user-id>"' on OSGi console
 #########################################################################################
-```
+````
 
-So the user needs to open the shown URL in a web browser, login with his Withings credentials, confirm that openHAB is allowed to access his data and at the end he is redirected to a page on github. There the user finds the command `withings:finishAuthentication "<accountId>" "<verifier>" "<user-id>"` with filled parameters that is needed to finish the authentication.
+The user needs to open the given URL in a web browser, log in with his Withings
+credentials, and confirm that openHAB is allowed to access the user's data.  At
+the end of this process, the user will be redirected to a page on github. There
+the user finds the command `withings:finishAuthentication "<accountId>" "<verifier>" "<user-id>"`
+with pre-filled parameters that is needed to finish the authentication.
 
-The binding stores the OAuth tokens, so that the user does not need to login again. From this point the binding is successfully configured.
+The binding stores the OAuth tokens so that the user does not need to log in
+again. From this point, the binding is successfully configured.
 
 ### Advanced OAuth Configuration
 
-The Withings Binding uses a default application registration to request an OAuth token. The application belongs to the binding developer. If the user wants to use his own application, the binding can be configured to use another OAuth consumer key and consumer secret. Please read the Withings documentation how to register an application (http://oauth.withings.com/en/partner/dashboard). After the application was created, Withings will generate a consumer key and secret. The redirect url must be configured, too. To change the OAuth parameters for the Withings binding the user can specify the following values in the `services/withings.cfg`:
+The Withings Binding uses a default application registration to request an OAuth
+token. The application belongs to the binding developer. If the user wants to
+use his own application, the binding can be configured to use another OAuth
+consumer key and consumer secret. Please read the [Withings documentation](http://oauth.withings.com/en/partner/dashboard)
+on how to register an application. After the application is created, Withings
+will generate a consumer key and secret. The redirect URL must be configured,
+too.
 
-The OAuth configuration must be stored in the file `services/withings-oauth.cfg`.
-
-### Multiple accounts
-
-If one owns multiple withings accounts these accounts must be configured specifically. In addition to the above mentioned `services/withings.cfg` configuration, an `accountId` can be specified.
-
-```
-thomas.token=74c0e77021ef5be1ec8dcb4dd88c1xckusadwe92f9541c86799dcbbfcb8fc8b236
-thomas.tokensecret=25f1098209xns511711b3287288f90740ff45532cef91658c5043db0b0e0c851c
-peter.token=74c0e77021ef5be1ec8dcb4dd88c1xckusadwe92f9541c86799dcbbfcb8fc8b236
-peter.tokensecret=25f1098209xns511711b3287288f90740ff45532cef91658c5043db0b0e0c851c
-```
 
 ## Item Configuration
 
-To bind a measure value to an item, the measure type has to be defined in the item binding config. Withings data can be bound to `Number` items only. The syntax for a Withings binding is 
+To bind a measure value to an item, the measure type has to be defined in the
+item binding config. Withings data can be bound to `Number` items only. The
+syntax for a Withings item binding is:
 
-```
+````
 withings="<measure type>"
-``` 
+````
 
-The following table shows the measure types and units, that are supported by the binding:
+The following table shows the measure types and units that are supported by the binding:
 
-| Measure type | Binding Config | Unit |
-|--------------|----------------|------|
-| Weight       | weight         | kg   |
-| Height       | height         | meter |
-| Fat Free Mass | fat_free_mass | kg   |
-| Fat Ratio    | fat_ratio      | %    |
+| Measure Type    | Binding Config | Unit |
+|-----------------|----------------|------|
+| Weight          | weight         | kg   |
+| Height          | height         | meter |
+| Fat Free Mass   | fat_free_mass | kg   |
+| Fat Ratio       | fat_ratio      | %    |
 | Fat Mass Weight | fat_mass_weight | kg |
 | Diastolic Blood Pressure | diastolic_blood_pressure | mmHg |
 | Systolic Blood Pressure | systolic_blood_pressure | mmHg |
-| Heart Pulse | heart_pulse | bpm |
+| Heart Pulse     | heart_pulse | bpm |
 
 
 ## Examples
 
-The following snippet shows some sample bindings:
+### Examples of multiple account configuration
 
-```
+````
+thomas.userid=ImThomas
+thomas.token=74c0e77021ef5be1ec8dcb4dd88c1xckusadwe92f9541c86799dcbbfcb8fc8b236
+thomas.tokensecret=25f1098209xns511711b3287288f90740ff45532cef91658c5043db0b0e0c851c
+peter.userid=PeterPeter
+peter.token=74c0e77021ef5be1ec8dcb4dd88c1xckusadwe92f9541c86799dcbbfcb8fc8b236
+peter.tokensecret=25f1098209xns511711b3287288f90740ff45532cef91658c5043db0b0e0c851c
+````
+
+### Item configuration examples
+
+````
 Number Weight     "Weight  [%.1f kg]"     { withings = "weight" }
 Number FatRatio   "FatRatio [%.1f %%]"    { withings = "fat_ratio" }
 Number HeartPulse "HeartPulse [%d bpm]"   { withings = "heart_pulse" }
-```
+Number PHeight    "Height  [%d in]"       { withings = "peter:height" }
+````

--- a/bundles/binding/org.openhab.binding.withings/src/main/java/org/openhab/binding/withings/internal/api/WithingsAccount.java
+++ b/bundles/binding/org.openhab.binding.withings/src/main/java/org/openhab/binding/withings/internal/api/WithingsAccount.java
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2010-2016 by the respective copyright holders.
+ * Copyright (c) 2010-2017 by the respective copyright holders.
  *
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
@@ -36,10 +36,8 @@ import oauth.signpost.signature.HmacSha1MessageSigner;
 
 /**
  * This class holds the Withings Account Credentials. It can also load
- * and store the credentials the either the {@code openhab.cfg} (legacy)
- * or the {@code services/withings-ouath.cfg} file. Where to store values
- * is decided whether {@code openhab.cfg} contains already keys
- * starting with {@code withings-oauth} or not.
+ * and store the credentials to either the {@code openhab.cfg} (legacy)
+ * or the {@code services/withings.cfg} file.
  *
  * Note: We'd decided to reimplement {@code store} and {@code load} rather
  * than using the {@link java.util.Properties} class because our keys can
@@ -53,7 +51,7 @@ public final class WithingsAccount {
 
     private static final Logger logger = LoggerFactory.getLogger(WithingsAccount.class);
 
-    private static final String SERVICE_NAME = "withings-oauth";
+    private static final String SERVICE_NAME = "withings";
 
     private static final String CONFIG_DIR = "." + File.separator + "configurations";
 
@@ -126,7 +124,7 @@ public final class WithingsAccount {
 
         try {
             // store properties either to openhab.cfg (legacy) or
-            // services/withings-oauth.cfg
+            // services/withings.cfg
             if (isLegacyConfiguration()) {
                 file = new File(CONFIG_DIR + File.separator + "openhab.cfg");
                 // in legacy case each property has to be prefixed with
@@ -136,12 +134,13 @@ public final class WithingsAccount {
             }
 
             if (!file.exists()) {
+                logger.debug("Configuration file '{}' was not found. Creating it now...", file.getAbsolutePath());
                 file.getParentFile().mkdirs();
                 file.createNewFile();
             }
 
             // if an account different from the default account is used
-            // it get's prefixed separated by "."
+            // it gets prefixed separated by "."
             if (!WithingsAuthenticator.DEFAULT_ACCOUNT_ID.equals(accountId)) {
                 prefix += accountId + ".";
             }
@@ -154,15 +153,15 @@ public final class WithingsAccount {
 
             store(config, file);
 
-            logger.debug("Saved WithingsAccount to file '{}'.", file.getAbsolutePath());
+            logger.debug("Saved Withings account to file '{}'.", file.getAbsolutePath());
         } catch (IOException ioe) {
-            logger.error("Couldn't write WithingsAccount to file '{}'.", file.getAbsolutePath());
+            logger.warn("Couldn't write Withings account to file '{}'.", file.getAbsolutePath());
         }
     }
 
     /**
-     * Checks whether the Binding configuration has been store to
-     * openhab.cfg rather than the /services/withings-oauth.cfg dir
+     * Checks whether the Binding configuration has been stored to
+     * openhab.cfg rather than services/withings.cfg.
      * 
      * @return true, if there is a configuration key like "withings-oauth"
      *         in openhab.cfg and false in all other cases.
@@ -220,5 +219,4 @@ public final class WithingsAccount {
     public String toString() {
         return "WithingsAccount [userId=" + userId + ", token=" + token + ", tokenSecret=" + tokenSecret + "]";
     }
-
 }

--- a/bundles/binding/org.openhab.binding.withings/src/main/java/org/openhab/binding/withings/internal/api/WithingsAuthenticator.java
+++ b/bundles/binding/org.openhab.binding.withings/src/main/java/org/openhab/binding/withings/internal/api/WithingsAuthenticator.java
@@ -248,7 +248,9 @@ public class WithingsAuthenticator implements ManagedService {
                 } else if ("tokensecret".equals(configKeyTail)) {
                     account.tokenSecret = value;
                 } else {
-                    throw new ConfigurationException(configKey, "The given configuration key is unknown!");
+                    String msg = "The configuration key '" + configKey + "' is not recognized and will be ignored.";
+                    logger.debug(msg);
+                    throw new ConfigurationException(configKey, msg);
                 }
             }
 
@@ -268,7 +270,7 @@ public class WithingsAuthenticator implements ManagedService {
                 printAuthenticationInfo(accountId);
             } else {
                 logger.warn(
-                        "Configuration details of Account '{}' are invalid please check openhab.cfg or withings.cfg.",
+                        "Configuration details of Account '{}' are invalid. Please check the configuration.",
                         accountId);
             }
         }

